application/Modules/Mouse/Producers/MouseMoveEventProducer.cs

```diff
@@ -1,16 +1,94 @@
-﻿using System.ComponentModel.Composition;
+using System.Threading;
+using System.Windows;
 using MORR.Modules.Mouse.Events;
 using MORR.Shared.Events.Queue;
+using MORR.Shared.Utility;
 
 namespace MORR.Modules.Mouse.Producers
 {
     /// <summary>
     ///     Provides a single-writer-multiple-reader queue for MouseMoveEvent
     /// </summary>
-    [Export(typeof(MouseMoveEventProducer))]
-    [Export(typeof(IReadOnlyEventQueue<MouseMoveEvent>))]
     public class MouseMoveEventProducer : DefaultEventQueue<MouseMoveEvent>
     {
-        // TODO: Implement this
+        /// <summary>
+        ///     The mouse position in screen coordinates in the last period.
+        ///     This field will be initialized to the mouse position
+        ///     when the StartTimer() is called.
+        /// </summary>
+        private NativeMethods.POINT lastMousePosition;
+
+        /// <summary>
+        ///     A timer that records the mouse position at specific intervals.
+        /// </summary>
+        private Timer? mousePositionRecordingTimer;
+
+        /// <summary>
+        ///     The sampling rate of the mouse position capture, in Hz.
+        /// </summary>
+        private int SamplingRate { get; set; }
+
+        /// <summary>
+        ///     The minimal distance(computed with screen coordinates) a mouse move
+        ///     must reach in a period to be recorded.
+        ///     A mouse move with distance less than the Threshold will be ignored,
+        ///     in other words, a new MouseMoveEvent will not be generated and
+        ///     the mouse position will not be recorded.
+        /// </summary>
+        private int Threshold { get; set; }
+
+        /// <summary>
+        ///     Get the mouse position and create & enqueue corresponding event
+        ///     if the Threshold is reached.
+        /// </summary>
+        /// <param name="stateInfo">state object</param>
+        private void GetMousePosition(object stateInfo)
+        {
+            // get the current mouse position as Point
+            NativeMethods.POINT currentMousePosition;
+            NativeMethods.GetCursorPos(out currentMousePosition);
```

Comment:
You can inline the declaration - e.g. `NativeMethods.GetCursorPos(out var currentMousePosition)`